webapp/src/Service/ExternalContestSourceService.php

```diff
@@ -1017,7 +1017,7 @@ protected function validateAndUpdateProblem(Event $event, EventData $data): void
             );
             $contestProblem->setShortname($data->label);
         }
-        if ($contestProblem->getColor() !== ($data->rgb)) {
+        if ($data->rgb && $contestProblem->getColor() !== $data->rgb) {
```

Comment:
Why is this change made? The RGB data is nullable and should be in the notification if it is not null?

Reply:
PC2 sends `color` but not `rgb`. In that case we made everything white, which doesn't seem OK. We **could** use `color` in that case, but that would change our colors (more than) slightly, so that didn't feel good either?

Reply:
If I read https://ccs-specs.icpc.io/2023-06/contest_api#problems correctly, the presence of `color` should imply the existence of the `rgb` value, as it is merely the human-readable description of the `rgb` field.

Reply:
I don't think anything in that file specifies either or none are required.

https://ccs-specs.icpc.io/2023-06/ccs_system_requirements#contest-api does however. I will check with PC^2, but I guess they won't be sending it anytime soon.

Reply:
Regardless, sending rgb *is* optional in the spec, so I'd say if we get no color but we do have one, it's better to stick to our color

Reply:
If the CCS we are shadowing from would not support `rgb` but we do (which could be a valid scenario), I don't think it makes sense to remove our value. But maybe I'm in the minority here and if so I'm OK with reverting this change.

Reply:
Turns out it was a bug in the YAML (an unquoted rgb string starting with `#` which PC2 saw as a comment and we somehow_ parsed as a string, not sure how). I'm fine reverting this.